packages/js/data/src/payment-gateway-suggestions/actions.ts

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Internal dependencies
+ */
+import { ACTION_TYPES } from './action-types';
+import { PaymentGatewaySuggestionsState, OfflinePaymentGateway } from './types';
+
+export function getPaymentGatewaySuggestionsRequest(): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_REQUEST;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_REQUEST,
+	};
+}
+
+export function getPaymentGatewaySuggestionsSuccess(
+	paymentGatewaySuggestions: PaymentGatewaySuggestionsState
+): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_SUCCESS;
+	paymentGatewaySuggestions: PaymentGatewaySuggestionsState;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_SUCCESS,
+		paymentGatewaySuggestions,
+	};
+}
+
+export function getPaymentGatewaySuggestionsError( error: unknown ): {
+	type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_ERROR;
+	error: unknown;
+} {
+	return {
+		type: ACTION_TYPES.GET_PAYMENT_GATEWAY_SUGGESTIONS_ERROR,
+		error,
+	};
+}
+
+export function getOfflinePaymentGatewaysRequest(): {
+	type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_REQUEST;
+} {
+	return {
+		type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_REQUEST,
+	};
+}
+
+export function getOfflinePaymentGatewaysSuccess(
+	offlineGateways: OfflinePaymentGateway[]
+): {
+	type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_SUCCESS;
+	offlineGateways: OfflinePaymentGateway[];
+} {
+	return {
+		type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_SUCCESS,
+		offlineGateways,
+	};
+}
+
+export function getOfflinePaymentGatewaysError( error: unknown ): {
+	type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_ERROR;
+	error: unknown;
+} {
+	return {
+		type: ACTION_TYPES.GET_OFFLINE_PAYMENT_GATEWAYS_ERROR,
+		error,
+	};
+}
+
+export function enablePaymentGatewayRequest( gatewayId: string ): {
+	type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_REQUEST;
+	gatewayId: string;
+} {
+	return {
+		type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_REQUEST,
+		gatewayId,
+	};
+}
+
+export function enablePaymentGatewaySuccess(
+	gatewayId: string,
+	isOffline: boolean
+): {
+	type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_SUCCESS;
+	gatewayId: string;
+	isOffline: boolean;
+} {
+	return {
+		type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_SUCCESS,
+		gatewayId,
+		isOffline,
+	};
+}
+
+export function enablePaymentGatewayError(
+	gatewayId: string,
+	error: unknown
+): {
+	type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_ERROR;
+	gatewayId: string;
+	error: unknown;
+} {
+	return {
+		type: ACTION_TYPES.ENABLE_PAYMENT_GATEWAY_ERROR,
+		gatewayId,
+		error,
+	};
+}
+
+export function* enablePaymentGateway(
+	gatewayId: string,
+	isOffline: boolean,
+	ajaxUrl: string,
+	gatewayToggleNonce: string
+) {
+	// Dispatch the request action
+	yield enablePaymentGatewayRequest( gatewayId );
+
+	try {
+		// Use apiFetch for the AJAX request
+		yield apiFetch( {
+			url: ajaxUrl,
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded',
+			},
+			body: new URLSearchParams( {
```

Comment:
It is unusual, but it's the same as it's done on the current settings page, see https://github.com/woocommerce/woocommerce/blob/881c50e310a3aef01264263497e6843bdf67e7ab/packages/js/data/src/payment-settings/actions.ts#L119-L119. I wanted to re-use the endpoint rather than creating new REST API in wc-admin that does the same